hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/index/hbase/SparkHoodieHBaseIndex.java

```diff
@@ -480,6 +486,61 @@ private Integer getNumRegionServersAliveForTable() {
   @Override
   public boolean rollbackCommit(String instantTime) {
     // Rollback in HbaseIndex is managed via method {@link #checkIfValidCommit()}
+    synchronized (SparkHoodieHBaseIndex.class) {
```

Comment:
@hj2016 We don't fire deletes to Hbase to enable rollback, instead, we just check for whether that update is present or not, if that entry is not valid, the record key will be re-ingested and the entry will be overwritten. Can you talk about a use-case where you require to delete entries ?

Reply:
Of course, the checkIfValidCommit method will detect whether the index commitTime is a valid index, and it has no effect on writing data after rollback. Here, it is possible to write hbase index data incorrectly without deleting the last commit failure. But what I consider is to ensure that the data and the content of the index are consistent. Is it possible to add a configuration here for users to choose? Or is it better not to delete?

Reply:
@hj2016 The problem with firing deletes is that you end up putting extra load on the Hbase cluster for every rollback which is undesirable. The index and data should eventually be in sync always, let me explain how : 

1) Say the batch of records with 4 records (uuid1, uuid2, uuid3, uuid4) was inserted into the index but the batch failed.
2) Rollback will delete the data but let the index be
3) The next batch of records with 6 records (uuid1, uuid2, uuid3, uuid4, uuid5, uuid6) will now be retried. The HbaseIndex for the first 4 records will be overwritten and there won't be any dangling or remnant indexes. 
The one way this can happen is if you tried a batch of records, they failed and then you skipped those batch of records, which isn't a very common scenario.
If there is another use-case for which you want to delete the records from Hbase, we can consider having a config, let me know.

Reply:
ok, then I add a parameter to control, the default is not to delete. how do you feel?

Reply:
Okay, that makes sense to me, please go ahead with the change.

Reply:
@n3nash  I have changed it. Can you please confirm my pr  when you free?